tests/api2/test_012_api_key.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+
+# Author: Eric Turgeon
+# License: BSD
+# Location for tests into REST API of FreeNAS
+
+import contextlib
+import io
+import json
+import os
+import sys
+
+import requests
+
+apifolder = os.getcwd()
+sys.path.append(apifolder)
+from functions import POST, GET, DELETE, PUT, SSH_TEST, wait_on_job
+from auto_config import pool_name, scale, ha, password, user, ip
+
+
+@contextlib.contextmanager
+def api_key(allowlist):
+    results = POST('/api_key/', {'name': "Test API KEY", 'allowlist': allowlist})
+    assert results.status_code == 200, results.text
+    key = results.json()
+
+    try:
+        yield key["key"]
+    finally:
+        results = DELETE(f"/api_key/id/{key['id']}/")
+        assert results.status_code == 200, results.text
+
+
+def test_root_api_key_websocket():
```

Comment:
I believe convention here is to number API tests. Short docstring blurb explaining what's being tested may be helpful for QE.

Reply:
I believe they number API tests because their execution order matters (not a good practice IMO). This module's tests can be executed in any order.

Docstrings added.